src/InjectImageIntrinsics.cpp

```diff
@@ -74,21 +74,30 @@ class InjectImageIntrinsics : public IRMutator {
         args[0] = call->name;
         args[1] = Variable::make(Handle(), call->name + ".buffer");
         for (size_t i = 0; i < call_args.size(); i++) {
-            string d = int_to_string(i);
-            string min_name = name + ".min." + d;
-            string min_name_constrained = min_name + ".constrained";
-            if (scope.contains(min_name_constrained)) {
-                min_name = min_name_constrained;
-            }
-            string extent_name = name + ".extent." + d;
-            string extent_name_constrained = extent_name + ".constrained";
-            if (scope.contains(extent_name_constrained)) {
-                extent_name = extent_name_constrained;
-            }
 
-            Expr min = Variable::make(Int(32), min_name);
-            args.push_back(mutate(call_args[i]) - min);
-            args.push_back(Variable::make(Int(32), extent_name));
+            // If this is an ordinary dimension, insert a variable that will be
+            // subsequently defined by StorageFlattening to with the min and
+            // extent. Otherwise, add a default value for the padded dimension
+            if (i < call->args.size()) {
```

Comment:
sneaky, this is very close to the for-loop condition (call_args.size() vs call->args.size()) ... let's add a comment pointing out the difference, this almost tripped me up on review